lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputHelper.java

```diff
@@ -113,12 +110,9 @@ private int doInternalCacheRead(long position, byte[] bytes, int offset, int len
         int remainingLength = length;
         while (remainingLength > 0) {
             int bytesReadFromCache = readPageFromCache(position, bytes, offset, remainingLength);
-            if (bytesReadFromCache == 0) {
+            if (bytesReadFromCache <= 0) {
                 break;
```

Comment:
pre-existing: 

https://github.com/trinodb/trino/blob/0c1fdcb2a736673fad219ea34a7f16dc70104582/lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioInputHelper.java#L127

Are we supposed to record the cache when not reading anything from it?

Reply:
Some parts of the cache could be read right ? `bytesRead` would capture that.